src/test/ram_expression_equal_clone_test.cpp

```diff
@@ -0,0 +1,196 @@
+/*
+ * Souffle - A Datalog Compiler
+ * Copyright (c) 2019, The Souffle Developers. All rights reserved
+ * Licensed under the Universal Permissive License v 1.0 as shown at:
+ * - https://opensource.org/licenses/UPL
+ * - <souffle root>/licenses/SOUFFLE-UPL.txt
+ */
+
+/************************************************************************
+ *
+ * @file ram_expression_equal_clone_test.cpp
+ *
+ * Tests equal and clone function of RamExpression classes.
+ *
+ ***********************************************************************/
+
+#include "RamExpression.h"
+
+#include "test.h"
+
+namespace souffle {
+
+namespace test {
+
+TEST(RamIntrinsicOperator, CloneAndEquals) {
+    // ADD(number(1), number(2))
+    std::vector<std::unique_ptr<RamExpression>> a_args;
+    a_args.emplace_back(new RamNumber(1));
+    a_args.emplace_back(new RamNumber(2));
+    RamIntrinsicOperator a(FunctorOp::ADD, std::move(a_args));
+
+    std::vector<std::unique_ptr<RamExpression>> b_args;
+    b_args.emplace_back(new RamNumber(1));
+    b_args.emplace_back(new RamNumber(2));
+    RamIntrinsicOperator b(FunctorOp::ADD, std::move(b_args));
+    EXPECT_EQ(a, b);
+    EXPECT_NE(&a, &b);
+
+    RamIntrinsicOperator* c = a.clone();
+    EXPECT_EQ(a, *c);
+    EXPECT_NE(&a, c);
+    delete c;
+
+    // NEG(number(1))
```

Comment:
Make this a separate test. `RamIntrinsicOperator2` or  `RamIntrinsicOperatorNeg`